src/components/structures/RoomView.js

```diff
@@ -143,17 +150,46 @@ module.exports = React.createClass({
             }
         });
 
+        if (this.props.roomAddress[0] == '#') {
+            // we always look up the alias from the directory server:
```

Comment:
in the case that `roomAddress` is an alias on a remote server, `MatrixClientPeg.get().getRoomIdForAlias` will fail (I think?) and we will not set `state.roomId` or call `onRoomIdResolved`.